help center: Rewrite /help/search-for-messages.

alya opened the pull request:
Goals:

- Organize search terms into categories.
- To avoid technical-sounding jargon, use the term "filter" rather "operator".
- Be more specific and consistent in describing search queries.
- Change how user search is described, as we now show user name pills.

Questions:

1. Do we have a standard for how we format example stream names? I went with bold here.
2. Thoughts on what this page should be called? We have "Advanced search" in the left sidebar, "Searching for messages" on the page itself, and "search-for-messages" in the URL. Perhaps we should align these better?

Notes:
- Once this PR is settled, we should update the `?` menu to line it up with the ordering and terminology on this page.
- See also [related CZO discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/.3F.20menu/near/1470680) (the output from which can be a follow-up).

Testing:
- Links were manually tested.

**Current page:** https://zulip.com/help/search-for-messages

(The page is very tall, so I'm not including a screenshot here.)

alya commented:
@drrosa @laurynmm It would be great to get your reviews on this one!

drrosa commented on templates/zerver/help/search-for-messages.md:
```diff
@@ -3,69 +3,125 @@
 Zulip has a powerful search engine under its hood. Search for messages using
 the search bar at the top of the app.
```
Maybe we should tweak the first couple of sentences and say something like:

```suggestion
Zulip has a powerful search engine under its hood. You can search for messages using
keywords and filters via the search bar at the top of the app.
```

laurynmm replied:
I was also wondering if adding a keyboard tip for the search shortcuts would make sense here (or elsewhere on the page, maybe at the end of the keyword search section?).

```
!!! keyboard_tip ""

    You can activate the search bar by typing <kbd>/</kbd> or
    <kbd>Ctrl</kbd> + <kbd>K</kbd>.
```

drrosa commented on templates/zerver/help/search-for-messages.md:
```diff
@@ -3,69 +3,125 @@
-the rule.  For example, `stream:design -is:resolved -has:image` will
-match all messages that don't contain an image and were sent to
-unresolved topics in the `#design` stream.
+### Negated filters
```
Maybe we should consider "Exclude filters" here?  I think it's a more common term elsewhere and it would sound a bit less technical compared to "Negated filters".

laurynmm replied:
"Exclude" is the verb we use in the search suggestions dropdown menu.

drrosa reviewed (commented):
> 1. Do we have a standard for how we format example stream names? I went with bold here.

I ran a search in our user docs and only found 2 examples. One of the examples uses bold formatting and the other one doesn't. I think it makes sense to use bold here and to standardize it.

> 2. Thoughts on what this page should be called? We have "Advanced search" in the left sidebar, "Searching for messages" on the page itself, and "search-for-messages" in the URL. Perhaps we should align these better?

I think it's best if the title in the left sidebar and the page itself always match up. My vote would go to "Searching for messages", but maybe it's worth considering a new title that includes the word "filter"?

I left two more comments below, but other than that I think everything looks good to me.

laurynmm commented on templates/zerver/help/search-for-messages.md:
```diff
@@ -74,31 +130,22 @@ query using the `stream:` or `streams:` operators will search all
 messages that you have access to in the selected stream(s).  For
```
This is the only remaining instance of "operators" on the page. Do we want to keep it or change it to "filters"?

I will note that we've recently linked to this section from the [API documentation](https://zulip.com/api/construct-narrow) as it's a helpful explanation for folks using narrows via the API, which does use the more technical term of "operator".

alya replied:
Changed to "filters", even though @timabbott  might complain.

timabbott replied:
I don't think the API docs mismatching this is an issue; nobody will be confused by it. I think `using the `stream:` or `streams:` filter type will search`  might be clearer, but I don't see anyone being confused by this as written.

laurynmm reviewed (commented):
This revision also looks and reads great to me! A definite improvement on the current documentation.

I added my feedback to @drrosa's and had one more small note about the last use of "operators" in the text.

Re: the questions
1. Formatting for stream names: I agree that bold with the `#` makes sense. For topic names (e.g. [the second note in this article](https://zulip.com/help/rename-a-stream)), we've been using quotes.
2. Consistent naming: I also vote for **Searching for messages** to be the sidebar and article title. I think updating the file name to be consistent with that makes sense too.

alya commented:
> I ran a search in our user docs and only found 2 examples. One of the examples uses bold formatting and the other one doesn't. I think it makes sense to use bold here and to standardize it.

Thanks! Where did you find the one that's not bold, so that I can update it?

alya commented:
Thanks for the feedback @drrosa and @laurynmm !

I'm skipping this part to not spend the time, but have otherwise addressed the comments above.

> I think updating the file name to be consistent with that makes sense too.

alya commented:
I also added a not about the format for stream and topic names to https://zulip.readthedocs.io/en/latest/documentation/helpcenter.html#user-interface.

drrosa commented:
> Thanks! Where did you find the one that's not bold, so that I can update it?

`help/setting-up-zulip-for-a-class.md`

<img width="596" alt="image" src="https://user-images.githubusercontent.com/2343554/205378891-4ecf8ab8-53d2-44a0-a03d-9f5c1de0fddd.png">

alya commented:
> `help/setting-up-zulip-for-a-class.md`

> I also noticed that all stream names in this page have `\#` escaped characters. We should probably remove the `\` character?

Thanks! Pushed a commit making those fixes.

<img width="693" alt="Screen Shot 2022-12-02 at 2 50 25 PM" src="https://user-images.githubusercontent.com/2090066/205402903-8b2b13fb-2614-43e9-8837-18a07fa68ac4.png">
<img width="806" alt="Screen Shot 2022-12-02 at 2 50 11 PM" src="https://user-images.githubusercontent.com/2090066/205402904-a837a3a0-ef8e-432e-816e-ff3684061fc2.png">

alya commented:
@timabbott I've addressed the feedback from @laurynmm and @drrosa  (other than renaming searchi-for-messages.md -> searching-for-messages.md, which I think we can skip).

Please take a look! Getting this PR finalized/merged will unblock us for figuring out parallel reorg improvements in the `?` menu.

timabbott commented on templates/zerver/help/search-for-messages.md:
```diff
@@ -1,104 +1,150 @@
 
-Zulip has a powerful search engine under its hood. Search for messages using
-the search bar at the top of the app.
+Zulip has a powerful search engine under its hood. You can search for messages using
```
> Zulip has a powerful search engine under its hood.

We might want to consider rewriting this phrase in a follow-up pass; it looks like we wrote that back in 2018 (6880af6c2e3b4e598f385e3c2efb2eba4e2ce226).
